x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -783,7 +799,7 @@ export class TaskRunner<
 
         return { interval: retryInterval };
       }),
-      monitoring,
+      monitoring: this.ruleMonitoring.getMonitoring(),
```

Comment:
After 1 execution, my rule saved object looks like:

```
{
	... other fields
	"executionStatus": {
		"status": "active",
		"lastExecutionDate": "2022-11-03T13:09:15.575Z",
		"error": null,
		"warning": null,
		"lastDuration": 7270
	},
	"monitoring": {
		"run": {
			"history": [{
				"duration": 7270,
				"success": true,
				"timestamp": 1667480955574
			}],
			"calculated_metrics": {
				"success_ratio": 1,
				"p99": 7270,
				"p50": 7270,
				"p95": 7270
			},
			"last_run": {
				"timestamp": "2022-11-03T13:09:15.574Z",
				"metrics": {
					"duration": 0,
					"total_search_duration_ms": null,
					"total_indexing_duration_ms": null,
					"total_alerts_detected": null,
					"total_alerts_created": null,
					"gap_duration_s": null
				}
			}
		}
	},
	"lastRun": {
		"alertsCount": {
			"new": 5,
			"ignored": 0,
			"recovered": 0,
			"active": 5
		},
		"outcomeMsg": null,
		"warning": null,
		"outcome": "succeeded"
	},
}
```

The `monitoring.run.last_run.duration` field doesn't look like it's been updated to the last duration.

Reply:
I'm sure this was covered in the RFC and I'm behind, but why a separate `warning` and `outcomeMsg` field? The `outcome` can be `succeeded | failed | warning` so I would expect any warnings to show up as `outcome: 'warning'`, `outcomeMsg: <warning message>`?

Reply:
Ah yes good catch WRT the duration, will fix that. 

From the RFC, the `warning` field is described to act sort of like a decorator to the outcome. My understanding is the outcome could be successful but there might have been a special flag raised in the rule run that should be noted. So in this case the `warning` field gets mapped to `reasons`. @XavierM could correct me if I'm wrong here.

Reply:
++